Kudu.Services/Deployment/PushDeploymentController.cs

```diff
@@ -50,26 +50,48 @@ public class PushDeploymentController : ApiController
         {
             using (_tracer.Step("ZipPushDeploy"))
             {
-                var zipFileName = Path.ChangeExtension(Path.GetRandomFileName(), "zip");
-                var zipFilePath = Path.Combine(_environment.ZipTempPath, zipFileName);
-
-                using (_tracer.Step("Writing zip file to {0}", zipFilePath))
+                var deploymentInfo = new ZipDeploymentInfo(_environment, _traceFactory)
                 {
-                    using (var file = FileSystemHelpers.CreateFile(zipFilePath))
-                    {
-                        await request.Content.CopyToAsync(file);
-                    }
-                }
+                    AllowDeploymentWhileScmDisabled = true,
+                    Deployer = deployer,
+                    IsContinuous = false,
+                    AllowDeferredDeployment = false,
+                    IsReusable = false,
+                    TargetChangeset = DeploymentManager.CreateTemporaryChangeSet(message: "Deploying from pushed zip file"),
+                    CommitId = null,
+                    RepositoryType = RepositoryType.None,
+                    Fetch = LocalZipFetch,
+                    DoFullBuildByDefault = false,
+                    Author = author,
+                    AuthorEmail = authorEmail,
+                    Message = message
+                };
+
+                return await PushDeployAsync(deploymentInfo, isAsync);
+            }
+        }
 
+        [HttpPost]
+        public async Task<HttpResponseMessage> WarPushDeploy(
+            HttpRequestMessage request,
+            [FromUri] bool isAsync = false,
+            [FromUri] string author = null,
+            [FromUri] string authorEmail = null,
+            [FromUri] string deployer = DefaultDeployer,
+            [FromUri] string message = DefaultMessage)
+        {
+            using (_tracer.Step("WarPushDeploy"))
+            {
                 var deploymentInfo = new ZipDeploymentInfo(_environment, _traceFactory)
                 {
                     AllowDeploymentWhileScmDisabled = true,
                     Deployer = deployer,
+                    TargetPath = @"webapps\ROOT",
```

Comment:
Cross-plat: Use Path.Combine, or Path.DirectorySeparatorChar if you want to concat strings yourself

Reply:
Good point, will update the PR!